### PR TITLE
[improvements] External Pages For Testimonials

### PR DESCRIPTION
**Pull request by CaneteJoash**
## Related Links
- Asana: https://app.asana.com/0/1202252431396579/1202253254120925
- Notion: https://www.notion.so/Project-UI-UX-Capstone-716c808e203c4cce85691202f6bd1eca?p=53ec6cc41557498b8e6bd53d53bbffa5

- Route Link:
http://localhost:3000/testimonial-content/Joash

## Description
- Create external pages for the testimonials for each review

## Notes

N/A

## Screenshots
image
image
image

### CODE REVIEW

**Comment by ume45876139** [comment]
# Comments

<details>
<summary>PC</summary>

https://user-images.githubusercontent.com/39412730/168258934-2b155d06-bf46-4080-97a4-5e307390078a.mp4

</details>

<details>
<summary>Tablet</summary>

https://user-images.githubusercontent.com/39412730/168259083-9130eab5-6286-4564-966d-b29ecdbccb5a.mp4


</details>


<details>
<summary>iPhone</summary>

IMG_8131


</details>

- [ ] Chat box should be displayed at testimonial external pages
- [ ] Title text is over lapping at iPhone

# Testing data
test device: Chrome, Safari
test time: 0.5h

**Review by ume45876139 — APPROVED** [review]
# Working

<details>
<summary>PC</summary>


https://user-images.githubusercontent.com/39412730/168266654-a6d17380-ee72-4afd-8682-7c10dd68f0f1.mp4


</details>

<details>
<summary>Tablet</summary>


https://user-images.githubusercontent.com/39412730/168266708-05ef11ea-205d-4229-8605-9b7d3786a3fb.mp4


</details>

<details>
<summary>iPhone</summary>

https://user-images.githubusercontent.com/39412730/168266977-43fd9657-3791-4ae6-969c-99063590eb0a.mp4

</details>

- No deformation
- Chat box is displayed
- Image is proper
- Text is proper

# Testing data
test device: Chrome, Safari
test time: 0.25h
comment: 3
iteration: 1
total test time: 0.75h

File: src/pages/components/testimonial/index.js [hunk 1]
```diff
@@ -1,16 +1,21 @@
-import React, { useRef } from 'react';
+import React, { useRef, useState } from 'react';
```
Comment (ErickDoroy) [inline src/pages/components/testimonial/index.js]: Remove useState if not use

File: src/pages/components/externalPages/TestimonialHeaderContent.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import { Link } from 'react-router-dom';
+import DownloadButton from '../DownloadButton';
+
+const TestimonialHeaderContent = ({ item }) => {
+  return (
+    <figure className='relative pb-5 xl:h-[595px] lg:h-[572px] md:h-[550] sm:h-[450px] h-[572px]'>
+      <img
```
Comment (ErickDoroy) [inline src/pages/components/externalPages/TestimonialHeaderContent.js]: Put an alt name like alt='header '

File: src/pages/components/externalPages/TestimonialMainContent.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            {item.isH4Tag && (
+              <h4 className='pt-4 text-xl font-semibold'>{item.text}</h4>
+            )}
+            {item.isH4PTag && <p className='pb-4 text-lg'>{item.text}</p>}
+            {item.isPTag && <p className='py-4 text-lg'>{item.text}</p>}
+            {item.isUlTag && (
+              <ul className='list-disc pl-8 pb-4'>{listItem}</ul>
+            )}
+            {item.isImgTag && (
+              <img
```
Comment (ErickDoroy) [inline src/pages/components/externalPages/TestimonialMainContent.js]: Put an alt name. you can change tag to this ``

**Review by ErickDoroy — APPROVED** [review]

File: src/pages/components/testimonial/index.js [hunk 4]
```diff
@@ -53,19 +58,27 @@ const Testimonial = () => {
...
+            (item, index) =>
+              name !== item.author && (
+                <Link
+                  to={`/testimonial-content/${item.author}`}
+                  target='none'
+                >
+                  <TestimonialCard key={index} item={item} />
+                </Link>
+              )
+          )}
```
Comment (egiegarciano14) [inline src/pages/components/testimonial/index.js]:
```suggestion
          {testimonials.map(
            (item, index) =>
              name !== item.author && (
                <Link
                  key={index}
                  to={`/testimonial-content/${item.author}`}
                  target='none'
                >
                  <TestimonialCard item={item} />
                </Link>
              )
          )}
```
There's a console error in key props, pls put the key={index} in the Link component

File: src/index.js [hunk 5]
```diff
@@ -28,6 +29,7 @@ root.render(
               path='/terms-and-conditions'
               element={<TermsAndConditions />}
             />
+            <Route path='/testimonial-content/:name' element={<TestimonialContent />} />
```
Comment (egiegarciano14) [inline src/index.js]:
```suggestion
            <Route path='/swipet-journeys/:name' element={<TestimonialContent />} />
```

File: src/pages/components/testimonial/index.js [hunk 6]
```diff
@@ -53,19 +58,27 @@ const Testimonial = () => {
...
           {...settings}
         >
-          {testimonials.map((item, index) => (
-            <TestimonialCard key={index} item={item} />
-          ))}
+          {testimonials.map(
+            (item, index) =>
+              name !== item.author && (
+                <Link
+                  to={`/testimonial-content/${item.author}`}
```
Comment (egiegarciano14) [inline src/pages/components/testimonial/index.js]:
```suggestion
                  to={`/swipet-journeys/${item.author}`}
```
Can we change the path to /swipet-journeys/ ?

File: src/pages/components/externalPages/TestimonialContentTemplate.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
+import React, { Fragment } from 'react';
+import TestimonialMainContent from './TestimonialMainContent';
+
+const TestimonialContentTemplate = ({ description }) => {
+
+  return (
+    <Fragment>
+      <header className='flex pt-[64px] sm:pt-[38px] px-[20px] sm:px-[55px] lg:pt-[51px] lg:px-[100px] xl:px-[200px] 2xl:px-[339px]'>
+        <img src='/images/Avatar01.svg' alt='profile' className='h-[71px]' />
```
Comment (egiegarciano14) [inline src/pages/components/externalPages/TestimonialContentTemplate.js]:
```suggestion
        <LazyLoadImage src='/images/Avatar01.svg' alt='profile' className='h-[71px]' />
```
Change the img tag to LazyLoadImage component
Dont forget also to import the LazyLoadImage

File: src/pages/components/externalPages/TestimonialHeaderContent.js [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+        src={item.src}
+        className='object-cover lg:object-[center_-400px] md:object-top h-full w-full'
+        alt={item.name}
+      />
+      <figcaption className='absolute left-0 right-0 bottom-5 bg-gradient-to-r from-orange-900 h-full'>
+        <div className='pt-[64px] sm:pt-[38px] px-[20px] sm:px-[55px] lg:pt-[51px] lg:px-[100px] lg:h-[572px] xl:px-[200px] 2xl:px-[339px]'>
+          <Link to='/'>
+            <div className='w-[165px] h-[39px] sm:m-auto lg:w-[192px] lg:h-[48px] hover:cursor-pointer'>
+              <img src='/images/LogoMix.svg' alt='swipet logo' />
+            </div>
```
Comment (egiegarciano14) [inline src/pages/components/externalPages/TestimonialHeaderContent.js]:
```suggestion
  return (
    <figure className='relative pb-5 xl:h-[595px] lg:h-[572px] md:h-[550] sm:h-[450px] h-[572px]'>
      <LazyLoadImage
        src={item.src}
        className='object-cover lg:object-[center_-400px] md:object-top h-full w-full'
        alt={item.name}
      />
      <figcaption className='absolute left-0 right-0 bottom-5 bg-gradient-to-r from-orange-900 h-full'>
        <div className='pt-[64px] sm:pt-[38px] px-[20px] sm:px-[55px] lg:pt-[51px] lg:px-[100px] lg:h-[572px] xl:px-[200px] 2xl:px-[339px]'>
          <Link to='/'>
            <div className='w-[165px] h-[39px] sm:m-auto lg:w-[192px] lg:h-[48px] hover:cursor-pointer'>
              <LazyLoadImage src='/images/LogoMix.svg' alt='swipet logo' />
            </div>
```

File: src/pages/components/externalPages/TestimonialMainContent.js [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
```
Comment (egiegarciano14) [inline src/pages/components/externalPages/TestimonialMainContent.js]:
```suggestion
              <LazyLoadImage
```

Comment (egiegarciano14) [inline src/pages/components/externalPages/TestimonialMainContent.js]:
```suggestion
              <LazyLoadImage
```

Comment (egiegarciano14) [inline src/pages/components/externalPages/TestimonialMainContent.js]:
```suggestion
              <LazyLoadImage
```

Comment (egiegarciano14) [reply src/pages/components/externalPages/TestimonialMainContent.js]: Pls double check other pages too :bow:

File: src/pages/components/testimonial/index.js (hunk 4 above)
Comment (egiegarciano14) [reply src/pages/components/testimonial/index.js]: Pls double check if there is still error in the console :bow:

**Comment by egiegarciano14** [comment]
Pls add more padding/margin
image

same as this 
image

**Comment by roseaugusto** [comment]
@CaneteJoash 
1. Limit the clickable area to image only **(exclude the gaps)**
image

2. Put in one folder the external pages all about the testimonials
image

**Review by roseaugusto — APPROVED** [review]
Approved!

File: src/pages/components/testimonial/TestimonialCard.js [hunk 10]
```diff
@@ -1,21 +1,25 @@
...
-      {/* Content */}
-      <figcaption className='absolute left-0 right-0 bottom-5 mr-5 rounded-lg bg-gradient-to-t from-black pb-5 text-white'>
-        <p className='mx-4 mb-4 text-sm font-semibold xl:text-lg'>
-          {item.text}
-        </p>
-        <p className='mx-4 font-allison text-[32px]'>{item.author}</p>
-      </figcaption>
+      <Link to={`/swipet-journeys/${item.author}`} target='none'>
+        {/* Image */}
+        <LazyLoadImage
```
Comment (egiegarciano14) [inline src/pages/components/testimonial/TestimonialCard.js]:
```suggestion
        <img
```

pwede ra ni <img tag, kay sa settings niya nag gamit naman ug lazyload

image

**Review by egiegarciano14 — APPROVED** [review]

**Pull request merged**